src/test.js

```diff
@@ -719,15 +727,15 @@ function internalStart( test ) {
 
 function numberOfTests( module ) {
 	var count = module.tests.length;
-	while ( module = module.childModule ) {
+	while ( ( module = module.childModule ) ) {
```

Comment:
Yeah, this is one I probably should have called out in my initial writeup.

ESLint has a rule `no-cond-assign`, which flags against assignments in conditional parts of a control statement (the assumption being the developer probably meant to use `==` or `===`). That rule can be configured to ignore such assignments if they are in redundant parentheses. In other words, a developer can signal his/her intention to assign in a condition by adding parentheses. (It looks a lot more reasonable without spaces inside parentheses, though.)

So we have three options here:
* Have ESLint ignore the violation on this line, but leave the rule on and don't use extra parentheses
* Use extra parentheses (this is what is happening now in my PR)
* Disable the rule, if we trust that developers know what they are doing in this space

How would you like me to proceed?